plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -82,190 +82,414 @@ function executeCLI($scriptName, &$args) {
-						$submissionFile->setGenreId($genre->getId());
-						$submissionFile->setFileStage(SUBMISSION_FILE_PROOF);
-						$submissionFile->setDateUploaded(Core::getCurrentDate());
-						$submissionFile->setDateModified(Core::getCurrentDate());
-						$submissionFile->setAssocType(ASSOC_TYPE_REPRESENTATION);
-						$submissionFile->setAssocId($publicationFormatId);
-						$submissionFile->setFileType('application/pdf');
-
-						// Assume open access, no price.
-						$submissionFile->setDirectSalesPrice(0);
-						$submissionFile->setSalesType('openAccess');
-
-						$submissionFileDao->insertObject($submissionFile, $temporaryFilename);
-						$fileManager = new FileManager();
-						$fileManager->deleteByPath($temporaryFilename);
-
-						echo __('plugins.importexport.csv.import.submission', array('title' => $title)) . "\n";
-					} else {
-						echo __('plugins.importexport.csv.unknownLocale', array('locale' => $locale)) . "\n";
-					}
-				} else {
-					echo __('plugins.importexport.csv.unknownPress', array('pressPath' => $pressPath)) . "\n";
+			$submissionId = $submissionDao->insertObject($submission);
+
+			$publication = $publicationDao->newDataObject();
+			$publication->setData('submissionId', $submissionId);
+			$publication->setData('version', 1);
+			$publication->setData('status', STATUS_PUBLISHED);
+			$publication->setData('datePublished', Core::getCurrentDate());
+			$publicationId = $publicationDao->insertObject($publication);
+
+			$submission->setData('currentPublicationId', $publicationId);
+			$submissionDao->updateObject($submission);
+
+			$contactEmail = $press->getContactEmail();
+			$authorsString = explode(";", $data->authorString);
+
+			foreach ($authorsString as $index => $authorString) {
+				// Examine the author string. Best case is: "Given1,Family1,email@address.com;Given2,Family2,email@address.com", etc
+				// But default to press email address based on press path if not present.
+				$givenName = $familyName = $emailAddress = null;
+				$authorString = trim($authorString); // whitespace.
+				[$givenName, $familyName, $emailAddress] = explode(",", $authorString);
+
+				$givenName = trim($givenName);
+				$familyName = trim($familyName);
+
+				if (empty($emailAddress)) {
+					$emailAddress = $contactEmail;
+				}
+
+				$emailAddress = trim($emailAddress);
+
+				$author = $authorDao->newDataObject();
+				$author->setSubmissionId($submissionId);
+				$author->setUserGroupId($authorGroup->getId());
+				$author->setGivenName($givenName, $data->locale);
+				$author->setFamilyName($familyName, $data->locale);
+				$author->setEmail($emailAddress);
+
+				if (!$index) {
+					$author->setPrimaryContact(true);
+					$publication->setData('primaryContactId', $author->getId());
 				}
+
+				$author->setData('publicationId', $publicationId);
+				$authorDao->insertObject($author);
+			} // Authors done.
+
+			$sanitizedAbstract = PKPString::stripUnsafeHtml($data->abstract);
```

Comment:
It's better to insert the content as it is, the sanitization will be done at the presentation layer.

Reply:
That was a request from Michael, cause there will be another use cases for the abstract beyond the presentation layer. I asked him about this as well.

Reply:
And what was the outcome of your conversation? All the codebase assumes the cleanup will happen at the UI.